### DISCUSSION TITLE
server shutdown

### DISCUSSION THREAD

**Discussion by xxxxasasxas** [Minestom/Minestom#1260] [Help] [answered]
hey! how would I check when the server shutdowns/plugin disables?

**Comment by GreatWyrm** [comment, marked as answer]
To run something on server shutdown, use [SchedularManager.buildShutdownTask()](https://javadoc.minestom.net/net/minestom/server/timer/SchedulerManager.html#buildShutdownTask(java.lang.Runnable)).

For extensions (Minestom's equivalent of plugins), they have the [Extension.terminate()](https://javadoc.minestom.net/net/minestom/server/extensions/Extension.html#terminate()) method they can override.

**Reply by xxxxasasxas** [reply]
thanks
